### PR TITLE
Livraison module Rapid7 Nexpose

### PR DESCRIPTION
**Pull request by TomInverse**
# Description
   Rapid7 Nexpose modules allows PF to get informations from nexpose audits.

# Impacts
   Modules not finished, need the Lord Semaan Mojo touch.

# Issue
   Modules not finished, need the Lord Semaan Mojo touch.

# Delete branch after merge
   YES

# NEWS file entries
    conf/nexpose-responces.txt
    html/pfappserver/lib/pfappserver/Form/Config/Pfdetect/nexpose.pm 
    html/pfappserver/root/config/pfdetect/type/nexpose.tt
    lib/pf/detect/parser/nexpose.pm

## New Features
    Rapid7 Nexpose results interpretations
    Module configuration in the PF GUI

# Upgrade notes

## Changes to filtering engines

The "matches" operator of the filtering engines (VLAN filters, RADIUS filters, etc) is now matching in a case insensitive way. If you need to perform case sensitive matches, switch to using regular expressions using the regex operator.

### CODE REVIEW

**Comment by extrafu** [comment]
Some comments:
1- what is  conf/nexpose-responces.txt ?
2- I guess it should be written "responses"
3- it's missing Rapid7 documentation - ie., how to configure the Nexpose part to integrate with PacketFence

**Comment by TomInverse** [comment]
1- what is conf/nexpose-responces.txt ?
	The nexpose-responSes.txt is the file containing all the
responses possibilities that PF must be able to manage.
	They is stillwork to be done to simplifie this file.

2- I guess it should be written "responses"
	Yes
3- it's missing Rapid7 documentation - ie., how to configure the
Nexpose part to integrate with PacketFence.
	Waiting for my collegue to define the backbone of it,

File: docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+    See PacketFence_Administration_Guide-docinfo.xml for authors, copyright 
+    and license information.
+
+////
+
+About this Guide
+----------------
+
+This guide has been created in order to help the configuration of Rapid7 InsighVM Nexpose and PacketFence, to provide informations and actions about devices compliances before and during network access.
```
Comment (extrafu) [inline docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc]: This guide has been created in order to help you with the configuration of Rapid7 InsighVM Nexpose and PacketFence, to provide information and actions about devices compliance before and during network access.

File: docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+Assumptions
+-----------
+* You have a configured PacketFence server environment with working test equipment.
+* You have a Centos7 server, ready and updated, we will use it as the 'Rapid7 InsightVM' server.
```
Comment (extrafu) [inline docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc]: CentOS 7

File: docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+* You have created an account at https://www.rapid7.com/[Rapid7] and downloaded the https://www.rapid7.com/products/insightvm/[InsightVM] binaries.
+* You have, at least, a trial available licence.
```
Comment (extrafu) [inline docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc]: a trial license.

File: docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+Installation
+------------
+
+Step 1: Rapid7 Insight Configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+How to configure Rapid7 InsightVM Nexpose Server to send data to the PacketFence server in a way it can understand and process it.
+
+WARNING: Disable the firewalld service, the iptables services, the SELinux services, and ... others IDS services.
```
Comment (extrafu) [inline docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc]: ... ? be specific to not leave the reader in ambiguous state.

File: docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+What you will write:What Nexpose will understand
+10.0.0.0/24:10.0.0.1-10.0.0.254
+10.0.0.1-10.0.0.254:10.0.0.1-10.0.0.254
+:===
+
+* In *Exclude*, put all the assets you want to exclude from the previous ranges.
+* In the upper *Menu*, select *Templates* and define the scope of the audit you want.
+* In the *Menu*, select *Engines* and define the engine, depending on the licence and scope you have.
+
+How to sent Syslogs to PacketFence
```
Comment (extrafu) [inline docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc]: Sending syslog information to PacketFence

File: docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+
+*Notification Method*: Select _Syslog message_
+
+*Syslog Server*: "PacketFence cluster VIP or server IP for a standalone"  
+|===
+
+In the *Menu*, select now *Schedule*, and define the schedules of your audits.
+
+
+Step 2: PacketFence configuration
```
Comment (extrafu) [inline docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc]: Configuration

File: docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+"What need to be done on the PacketFence Server to understand and take action about endpoints containing threats"
```
Comment (extrafu) [inline docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc]: needs

File: docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+
+Get PacketFence ready to welcome Nexpose Syslogs
```
Comment (extrafu) [inline docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc]: PacketFence changes to handle Nexpose syslog information

File: docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+WARNING: If you are using a PacketFence cluster, you will need to do these steps in all your PacketFence servers.
```
Comment (extrafu) [inline docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc]: on all your

File: docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+
+* Logon to PacketFence Server with a _ssh_ terminal.
+* Create the fifo pipe file that PacketFence will use to get datas from Nexpose
```
Comment (extrafu) [inline docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc]: datas -> data

File: docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+ 
+ mkfifo /usr/local/pf/var/run/nexpose_pipe
+
+* Create a new file named /etc/rsyslog.d/nexpose-log.conf.
+
+ # rsyslog conf for Rapid7 Nexpose server logs reception
+ if $programname == 'Nexpose' then /usr/local/pf/var/run/nexpose_pipe
+ & ~
+
+* Modify /etc/rsyslog.conf to accept syslogs on 'udp 514'
```
Comment (extrafu) [inline docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc]: syslogs -> syslog data

File: docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc [hunk 12]
```diff
@@ -0,0 +1,185 @@
...
+
+ $ModLoad imudp
+ $UDPServerRun 514
+
+* Restart the 'rsyslog' service
+** Run this command:
+  
+  service rsyslog restart
+
+At this point Packetfence must be able to get the Nexpose audits results.
```
Comment (extrafu) [inline docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc]: PacketFence

Comment (extrafu) [reply docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc]: audit results

File: docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc [hunk 13]
```diff
@@ -0,0 +1,185 @@
...
+
+TIP: You can see if the Nexpose server is sending to the right server by minitoring both sides traffic, with a 'tcpdump -i any dst host YOUR_PACKETFENCE_SERVERI' on your Rapid7 Nexpose server and 'tcpdump -i any port 514' on the PacketFence server.
```
Comment (extrafu) [inline docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc]: monitoring

File: docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc [hunk 14]
```diff
@@ -0,0 +1,185 @@
...
+
+Get PacketFence to be able to configure what action need to be taken
```
Comment (extrafu) [inline docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc]: Integrating with PacketFence violation system

File: docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc [hunk 15]
```diff
@@ -0,0 +1,185 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Let's give PacketFence the Nexpose parser it needs to detect Nexpose events ::
+* Logon to PacketFence Server with a browser on the PacketFenceGUI
```
Comment (extrafu) [inline docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc]: server

Comment (extrafu) [reply docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc]: PacketFenceGUI -> PacketFence admin interface

File: docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc [hunk 16]
```diff
@@ -0,0 +1,185 @@
...
+* Click on 'SAVE'.
+
+Once done, we need to restart the good services ::
+* Restart the 'pfdetect' and the 'pfqueue' service.
+----
+/usr/local/pf/pfcmd service pfdetect restart
+/usr/local/pf/pfcmd service pfqueue restart
+----
+
+Now that PacketFence got the datas to process, lets tell it what to do with it ::
```
Comment (extrafu) [inline docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc]: 'got the datas to process, lets tell it what to do with it ::' ->

is properly configured to receive information from Nexpose, lets configure it to perform some actions:

File: docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc [hunk 17]
```diff
@@ -0,0 +1,185 @@
...
+* In the PacketFence GUI, go to *Configuration -> Compliance -> Violations*
```
Comment (extrafu) [inline docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc]: In the PacketFence GUI -> From the PacketFence admin interface

File: docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc [hunk 18]
```diff
@@ -0,0 +1,185 @@
...
+* Go down on the page to the end of the section and note the last violation ID XXXXXXX
+* Click on *ADD VIOLATION*
+
+|===
+|*Definition*
+|*Enable*: Set it to *ON* 
+
+*Identifier*: Use the previously noted violation ID and add 1 to it. (But you can use what ever you want.)
```
Comment (extrafu) [inline docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc]: what ever you want. => whatever you want

File: docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc [hunk 19]
```diff
@@ -0,0 +1,185 @@
...
+
+*Action*: This is where you put what you want PacketFence to do, in this case.
+|===
+
+
+|===
+|*Triggers*
+|Click on the plus *(+)*, on the right side of the page.
+
+On the seconde line, choose the appropriate trigger between "*nexpose_event_contains*" or "*nexpose_event_start_with*"
```
Comment (extrafu) [inline docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc]: second

File: docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc [hunk 20]
```diff
@@ -0,0 +1,185 @@
...
+
+Choose "nexpose_event_contains" if you know, for exemple, the "*Common Vulnerabilities and Exposures*" you want to take action on.
```
Comment (extrafu) [inline docs/PacketFence_Rapid7NexposeInsightVM_Quick_Install_Guide.asciidoc]: for example

**Comment by julsemaan** [comment]
Merged changes in feature/rapid7-scan, will open a PR with that branch

**Pull request closed without merging**